src/battle/overlay_12_0224E4FC.c

```diff
@@ -5541,3 +5542,842 @@ BOOL TryFling(BattleSystem *bsys, BATTLECONTEXT *ctx, int battlerId) {
+    int i;
+    s32 dmg = 0;
+    s32 dmg2 = 0;
+    u8 moveType;
+    u8 moveCategory;
+    u16 monAtk;
+    u16 monDef;
+    u16 monSpAtk;
+    u16 monSpDef;
+    s8 statChangeAtk;
+    s8 statChangeDef;
+    s8 statChangeSpAtk;
+    s8 statChangeSpDef;
+    u8 level;
+    u16 movePower;
+    u16 item;
+    u32 battleType;
+    MoveDamageCalc calcAttacker;
+    MoveDamageCalc calcTarget;
+    
+    GF_ASSERT(crit == 1 || crit > 1);
+    
+    monAtk = GetBattlerVar(ctx, battlerIdAttacker, BMON_DATA_ATK, NULL);
+    monDef = GetBattlerVar(ctx, battlerIdTarget, BMON_DATA_DEF, NULL);
+    monSpAtk = GetBattlerVar(ctx, battlerIdAttacker, BMON_DATA_SPATK, NULL);
+    monSpDef = GetBattlerVar(ctx, battlerIdTarget, BMON_DATA_SPDEF, NULL);
+    statChangeAtk = GetBattlerVar(ctx, battlerIdAttacker, BMON_DATA_STAT_CHANGE_ATK, NULL) - 6;
+    statChangeDef = GetBattlerVar(ctx, battlerIdTarget, BMON_DATA_STAT_CHANGE_DEF, NULL) - 6;
+    statChangeSpAtk = GetBattlerVar(ctx, battlerIdAttacker, BMON_DATA_STAT_CHANGE_SPATK, NULL) - 6;
+    statChangeSpDef = GetBattlerVar(ctx, battlerIdTarget, BMON_DATA_STAT_CHANGE_SPDEF, NULL) - 6;
+    level = GetBattlerVar(ctx, battlerIdAttacker, BMON_DATA_LEVEL, NULL);
+    calcAttacker.species = GetBattlerVar(ctx, battlerIdAttacker, BMON_DATA_SPECIES, NULL);
+    calcTarget.species = GetBattlerVar(ctx, battlerIdTarget, BMON_DATA_SPECIES, NULL);
+    calcAttacker.hp = GetBattlerVar(ctx, battlerIdAttacker, BMON_DATA_HP, NULL);
+    calcTarget.hp = GetBattlerVar(ctx, battlerIdTarget, BMON_DATA_HP, NULL);
+    calcAttacker.maxHp = GetBattlerVar(ctx, battlerIdAttacker, BMON_DATA_MAXHP, NULL);
+    calcTarget.maxHp = GetBattlerVar(ctx, battlerIdTarget, BMON_DATA_MAXHP, NULL);
+    calcAttacker.status = GetBattlerVar(ctx, battlerIdAttacker, BMON_DATA_STATUS, NULL);
+    calcTarget.status = GetBattlerVar(ctx, battlerIdTarget, BMON_DATA_STATUS, NULL);
+    calcAttacker.ability = GetBattlerAbility(ctx, battlerIdAttacker);
+    calcTarget.ability = GetBattlerAbility(ctx, battlerIdTarget);
+    calcAttacker.gender = GetBattlerVar(ctx, battlerIdAttacker, BMON_DATA_GENDER, NULL);
+    calcTarget.gender = GetBattlerVar(ctx, battlerIdTarget, BMON_DATA_GENDER, NULL);
+    calcAttacker.type1 = GetBattlerVar(ctx, battlerIdAttacker, BMON_DATA_TYPE_1, NULL);
+    calcTarget.type1 = GetBattlerVar(ctx, battlerIdTarget, BMON_DATA_TYPE_1, NULL);
+    calcAttacker.type2 = GetBattlerVar(ctx, battlerIdAttacker, BMON_DATA_TYPE_2, NULL);
+    calcTarget.type2 = GetBattlerVar(ctx, battlerIdTarget, BMON_DATA_TYPE_2, NULL);
+    
+    item = GetBattlerHeldItem(ctx, battlerIdAttacker);
+    calcAttacker.item = GetItemVar(ctx, item, ITEM_VAR_HOLD_EFFECT);
+    calcAttacker.mod = GetItemVar(ctx, item, ITEM_VAR_MODIFIER);
+    
+    item = GetBattlerHeldItem(ctx, battlerIdTarget);
+    calcTarget.item = GetItemVar(ctx, item, ITEM_VAR_HOLD_EFFECT);
```

Comment:
should this really be labelled as item? it's the hold effect from it, not the item itself